plugins/azure/server/index.ts

```diff
@@ -1,9 +1,14 @@
-import { PluginManager, PluginType } from "@server/utils/PluginManager";
+import { Hook, PluginManager } from "@server/utils/PluginManager";
 import config from "../plugin.json";
 import router from "./auth/azure";
 import env from "./env";
 
-PluginManager.register(PluginType.AuthProvider, router, {
-  ...config,
-  enabled: !!env.AZURE_CLIENT_ID && !!env.AZURE_CLIENT_SECRET,
-});
+const enabled = !!env.AZURE_CLIENT_ID && !!env.AZURE_CLIENT_SECRET;
+
+if (enabled) {
+  PluginManager.add({
+    ...config,
```

Comment:
So we should be able to finally get rid of these spreads now?

Reply:
We might still need `priority`, `name`, `description` etc to pass as params. Sorry, is the end goal to completely get rid of the JSON config and instead pass these explicitly wherever required?

Reply:
True on priority. My thinking with the `.json` is that it's a good safe way to share config when both the client and server need to share information. In the cases where a plugin doesn't have any client component then it's not _required_ to have a json

Reply:
>In the cases where a plugin doesn't have any client component then it's not required to have a json

I suppose then the `...config` doesn't hurt? Keep it as a convention?